server/broadcast.go

```diff
@@ -429,6 +486,11 @@ func transcodeSegment(cxn *rtmpConnection, seg *stream.HLSSegment, name string,
 		sess = newSess
 	}
 
+	cxn.sessManager.sessLock.Lock()
+	sess.SegsInFlight = append(sess.SegsInFlight, SegFlightMetadata{startTime: time.Now(), segDur: time.Duration(int64(seg.Duration*1e9)) * time.Nanosecond})
+	cxn.sessManager.sessLock.Unlock()
+	glog.Warningf("SUBMITTING SEG %v Session=%v InFlightStart=%v SegDur=%v", seg.SeqNo, sess.OrchestratorInfo.Transcoder, sess.SegsInFlight[len(sess.SegsInFlight)-1].startTime, sess.SegsInFlight[len(sess.SegsInFlight)-1].segDur)
```

Comment:
Possible race here since we read SegsInFlight without a lock? (Not a problem if this log is going to be removed anyway).

Reply:
Yep planning to remove the log before push!